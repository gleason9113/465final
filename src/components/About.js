import React from 'react';

// Renders "About" page
function About() {
  return (
    <div className="container mx-auto text-center">
      <h1 className="about display-1" data-testid="about">
        About
      </h1>
      <p data-testid="p1">
        This is a CS465/565 Final Project by Ariel, Cera, and Alex
      </p>
      <p data-testid="p2">
        We wanted to implement a game using Node.js, React.js, Express.js, and
        Socket.io
      </p>
      <p data-testid="p3">We decided on a Tic-Tac-Toe game</p>
      <p data-testid="p4">Our project code can be found in: </p>
      <p>
        <a
          href="https://github.com/gleason9113/465final"
          target="_blank"
          rel="noreferrer"
          data-testid="link-project"
        >
          Project
        </a>
      </p>
      <p>More about each of us can be found in:</p>
      <p>
        <a
          href="https://github.com/C3ra906"
          target="_blank"
          rel="noreferrer"
          data-testid="cera-git"
        >
          Cera's Github
        </a>
      </p>
      <p>
        <a
          href="https://github.com/sashetov"
          target="_blank"
          rel="noreferrer"
          data-testid="alex-git"
        >
          Alex's Github
        </a>
      </p>
      <p>
        <a
          href="https://github.com/gleason9113"
          target="_blank"
          rel="noreferrer"
          data-testid="ariel-git"
        >
          Ariel's Github
        </a>
      </p>
    </div>
  );
}

export default About;
