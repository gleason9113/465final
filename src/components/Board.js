import React, { useEffect, useRef, useState } from 'react';

const Board = ({ socket, first_player, second_player, newGame }) => {
  // Hooks
  const [boardState, setBoard] = useState(['', '', '', '', '', '', '', '', '']);
  const [message, setMessage] = useState('');
  const [update, setUpdate] = useState(0);

  // Variables and constants
  const reference = useRef(null);
  const markSquare = useRef(false);
  const errorCode = useRef(-1);
  const winner = useRef(-2);
  const button = useRef();

  // useEffect works after render
  useEffect(() => {
    console.log('socket:', socket);
    window.socket = socket;
  }, [socket]);

  // Functions
  newGame.current = false;

  // Sends the server which square was clicked. Server determines validity.
  const move = (index) => {
    // Sends server location player wants to mark
    let win = -2;
    setUpdate(update + 1);
    setMessage('');

    socket.emit('move', { move_id: index });

    // Captures status message and errorCode from server
    socket.on('move_done', (data) => {
      setUpdate(update + 1);
      console.log('From Server' + JSON.stringify(data));

      let state = [...boardState];

      // Receives status and errorCode from the server for use on client side
      if (data.status === 'error') {
        errorCode.current = data.errorCode;
        markSquare.current = false;
      } else if (data.status === 'success') {
        errorCode.current = 7; //success
        markSquare.current = true;
        win = data.data.gameWinner;
        state = [...data.data.boardState];
      } else {
        console.log('unknown error occured w/ client side move_done');
        errorCode.current = -1;
        markSquare.current = false;
      }
      mark(state, win);
    });
  };

  // Marks the square if appropriate and updates boardState
  const mark = (state, win) => {
    // Marking square allowed
    if (markSquare.current === true) {
      //re-render
      console.log('The move is valid');
      setUpdate(update + 1);
    }
    // Marking square not allowed
    else {
      console.log('The move is invalid');
      setUpdate(update + 1);
    }

    // Updates the board
    setBoard([...state]);
    console.log('Updated board:');
    console.log(state);

    // Determine user message to display
    message_picker();

    // Checks to see if a player won
    if (win !== -2 && win !== null) {
      if (win === -1) {
        winner.current = -1;
        console.log(
          'Error: One of the players is missing. Please join the game.'
        );
      } else if (win === 0) {
        winner.current = 0;
        console.log(`${first_player.current} won!`); // TODO: get player names to display
      } else if (win === 1) {
        winner.current = 1;
        console.log(`${second_player.current} won!`); // TODO: get player names to display
      } else if (win === 2) {
        winner.current = 2;
        console.log('Tied. Gameover');
      }

      winner_determined();
      // TODO: ASk to replay or go back to previous screen
    }

    errorCode.current = -1; // reset errorCode
    markSquare.current = false; // reset markSquare
  };

  // User messages after a move
  const message_picker = () => {
    console.log('in message');
    if (errorCode.current === 1) {
      setMessage(`You are not in a game. Please join.`);
    } else if (errorCode.current === 2) {
      setMessage(`Invalid move id`);
    } else if (errorCode.current === 3) {
      setMessage(`You do not have an opponent. Please wait.`);
    } else if (errorCode.current === 4) {
      setMessage(`That square is taken! Please try another square.`);
    } else if (errorCode.current === 5) {
      setMessage(`You can't play out of turn! Wait for your opponent to play.`);
    } else if (errorCode.current === 7) {
      setMessage(`Move was a success`);
    } else if (errorCode.current === -1) {
      setMessage(`Default error...hmm..something must be wrong`);
    } else {
      setMessage(``);
    }
  };

  // Updates user message if a game is over
  const winner_determined = () => {
    if (winner.current === -1) {
      setMessage('Error: One of the players is missing. Please join the game.');
    } else if (winner.current === 0) {
      setMessage(`${first_player.current} won!`);
    } else if (winner.current === 1) {
      setMessage(`${second_player.current} won!`);
    } else if (winner.current === 2) {
      setMessage('Tied. Gameover');
    } else {
      setMessage(``);
    }
  };

  // Handles Reset Button Click
  const handleReset = (event) => {
    event.preventDefault();

    // Reset all variables and states managed on front-end
    const empty_board = ['', '', '', '', '', '', '', '', ''];
    const restart = 'Please reset the game with same players and same game id';
    reference.curret = null;
    markSquare.current = false;
    errorCode.current = -1;
    winner.current = -2;
    newGame.current = true;

    setBoard([...empty_board]);
    setMessage(``);
    setUpdate(0);

    console.log('everything reset');
    console.log(
      `message: ${message} board: ${boardState} reference: ${reference.current} markSquare: ${markSquare.current} errorCode: ${errorCode.current} winner: ${winner.current}`
    );

    // Ask back-end to restart the game
    socket.emit('restart', restart);
  };

  // Buttons
  const reset_button = (
    <button
      className="col reset-button"
      id="reset_game"
      type="button"
      onClick={handleReset}
    >
      Reset Game
    </button>
  );

  const replay_button = (
    <button
      className="col reset-button"
      id="play_again"
      type="button"
      onClick={handleReset}
    >
      Play Again
    </button>
  );

  if (winner.current === 0 || winner.current === 1 || winner.current === 2) {
    button.current = replay_button;
  } else {
    button.current = reset_button;
  }

  // renders the board
  return (
    <div ref={reference} className="container-sm w-50">
      <div className="row">
        <div
          className="col cell text-center"
          id="0"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
          aria-label="cell 0"
        >
          {boardState[0]}
        </div>
        <div
          className="col cell text-center"
          id="1"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
        >
          {boardState[1]}
        </div>
        <div
          className="col cell text-center"
          id="2"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
          aria-label="cell 2"
        >
          {boardState[2]}
        </div>
      </div>
      <div className="row text-center">
        <div
          className="col cell text-center"
          id="3"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
        >
          {boardState[3]}
        </div>
        <div
          className="col cell text-center"
          id="4"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
        >
          {boardState[4]}
        </div>
        <div
          className="col cell text-center"
          id="5"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
        >
          {boardState[5]}
        </div>
      </div>
      <div className="row">
        <div
          className="col cell text-center"
          id="6"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
        >
          {boardState[6]}
        </div>
        <div
          className="col cell text-center"
          id="7"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
        >
          {boardState[7]}
        </div>
        <div
          className="col cell text-center"
          id="8"
          onClick={(event) => {
            // Allow clicking unless the game is over
            if (winner.current === -2) {
              move(event.target.id);
            }
          }}
        >
          {boardState[8]}
        </div>
      </div>
      <div className="row" style={{ margin: '20px' }}></div>
      <div className="row">
        <div
          className="col message text-center"
          id="user_message"
          aria-label="message board"
        >
          {message}
        </div>
      </div>
      <div className="row">{button.current}</div>
    </div>
  );
};

export default Board;
