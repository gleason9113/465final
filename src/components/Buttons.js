import React from 'react';

// set up buttons to navigate bewtween pages? Or we can do navbar. still thinking
function Buttons({ handleAbout, handleConnect, handleHowToPlay }) {
  return (
    <div className="nav-container mx-auto my-auto text-center">
      <button
        className="nav-btn"
        id="about"
        type="button"
        onClick={handleAbout}
      >
        About
      </button>
      <button
        className="nav-btn"
        id="connect"
        type="button"
        onClick={handleConnect}
      >
        Connect
      </button>
      <button
        className="nav-btn"
        id="how"
        type="button"
        onClick={handleHowToPlay}
      >
        How To Play
      </button>
    </div>
  );
}

export default Buttons;
