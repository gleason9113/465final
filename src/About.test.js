import {
  render,
  fireEvent,
  screen,
  within,
  userEvent,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import MockedSocket from 'socket.io-mock';
import About from './components/About';
import HowToPlay from './components/HowToPlay';
import Form from './components/Form';
import Home from './components/Home';
import Buttons from './components/Buttons';
import Board from './components/Board';
import Comments from './components/Comments';
import ButtonHandler from './components/ButtonHandler';

// Tests ButtonHandler render to match snapshot
test('ButtonHanddler render to match snapshot', () => {
  // Mock socket
  const socket = new MockedSocket();

  // render
  const buttonHandler = renderer
    .create(<ButtonHandler socket={socket} />)
    .toJSON();

  // expected results
  expect(buttonHandler).toMatchSnapshot();
});

// Tests Buttons
test('Buttons function correctly', () => {
  //this one passes
  // Mock functions
  const handleHome = jest.fn();
  const handleAbout = jest.fn();
  const handleConnect = jest.fn();
  const handleHowToPlay = jest.fn();
  const handleComments = jest.fn();

  // render About on virtual dom
  render(
    <Buttons
      handleHome={handleHome}
      handleAbout={handleAbout}
      handleConnect={handleConnect}
      handleHowToPlay={handleHowToPlay}
      handleComments={handleComments}
    />
  );

  // buttons to test
  const home = screen.getByTestId('home_button');
  const about = screen.getByTestId('about_button');
  const howTo = screen.getByTestId('howToPlay_button');
  const connect = screen.getByTestId('connect_button');
  const comment = screen.getByTestId('comment_button');

  // interaction
  fireEvent.click(home);
  fireEvent.click(about);
  fireEvent.click(howTo);
  fireEvent.click(connect);
  fireEvent.click(comment);

  // expected results
  expect(home).toBeDefined();
  expect(about).toBeDefined();
  expect(howTo).toBeDefined();
  expect(connect).toBeDefined();
  expect(comment).toBeDefined();

  expect(handleHome).toHaveBeenCalledTimes(1);
  expect(handleAbout).toHaveBeenCalledTimes(1);
  expect(handleHowToPlay).toHaveBeenCalledTimes(1);
  expect(handleConnect).toHaveBeenCalledTimes(1);
  expect(handleComments).toHaveBeenCalledTimes(1);
});

// Buttons render to match snapshot
test('Buttons render to match snapshot', () => {
  // Mock functions
  const handleHome = jest.fn();
  const handleAbout = jest.fn();
  const handleConnect = jest.fn();
  const handleHowToPlay = jest.fn();
  const handleComments = jest.fn();

  // render
  const buttons = renderer
    .create(
      <Buttons
        handleHome={handleHome}
        handleAbout={handleAbout}
        handleConnect={handleConnect}
        handleHowToPlay={handleHowToPlay}
        handleComments={handleComments}
      />
    )
    .toJSON();

  // expected results
  expect(buttons).toMatchSnapshot();
});

// Tests About page rendering
test('About page renders correctly', () => {
  // this one passes
  // render About on virtual dom
  render(<About />);

  // elements to test
  const heading = screen.getByTestId('about');
  const p1 = screen.getByTestId('p1');
  const p2 = screen.getByTestId('p2');
  const p3 = screen.getByTestId('p3');
  const p4 = screen.getByTestId('p4');
  const project = screen.getByTestId('link-project');
  const cera_git = screen.getByTestId('cera-git');
  const alex_git = screen.getByTestId('alex-git');
  const ariel_git = screen.getByTestId('ariel-git');

  // interaction
  fireEvent.click(project);
  fireEvent.click(cera_git);
  fireEvent.click(alex_git);
  fireEvent.click(ariel_git);

  // expected results
  expect(heading).toHaveTextContent('About');
  expect(p1).toHaveTextContent(
    'This is a CS465/565 Final Project by Ariel, Cera, and Alex'
  );
  expect(p2).toHaveTextContent(
    'We wanted to implement a game using Node.js, React.js, Express.js, and Socket.io'
  );
  expect(p3).toHaveTextContent('We decided on a Tic-Tac-Toe game');
  expect(p4).toHaveTextContent('Our project code can be found in:');
  expect(project).toHaveAttribute(
    'href',
    'https://github.com/gleason9113/465final'
  );
  expect(project).toHaveAttribute('target', '_blank');
  expect(cera_git).toHaveAttribute('href', 'https://github.com/C3ra906');
  expect(cera_git).toHaveAttribute('target', '_blank');
  expect(alex_git).toHaveAttribute('href', 'https://github.com/sashetov');
  expect(alex_git).toHaveAttribute('target', '_blank');
  expect(ariel_git).toHaveAttribute('href', 'https://github.com/gleason9113');
  expect(ariel_git).toHaveAttribute('target', '_blank');
});

// About render to match snapshot
test('About render to match snapshot', () => {
  // render
  const about = renderer.create(<About />).toJSON();

  // expected results
  expect(about).toMatchSnapshot();
});

// Tests How To Play page rendering
test('How To Play page renders correctly', () => {
  // render HowToPlay on virtual dom
  render(<HowToPlay />);

  // elements to test
  const heading = screen.getByTestId('how_to_header');
  const two = screen.getByTestId('two_players');
  const rules = screen.getByTestId('rules');
  const list = screen.getByTestId('list_of_rules');
  const { getAllByRole } = within(list); // issue in ariel's side
  const rule_list = getAllByRole('listitem');
  const wins = screen.getByTestId('wins');
  const ties = screen.getByTestId('ties');

  // expected results
  expect(rule_list.length).toBe(9); // possible issue
  expect(heading).toHaveTextContent('How to Play Tic-Tac-Toe');
  expect(two).toHaveTextContent('This game is played with two players');
  expect(rules).toHaveTextContent('Rules:');
  expect(wins).toHaveTextContent('Example Wins:');
  expect(ties).toHaveTextContent('Example Tied Conditions:');
});

// HowToPlay render to match snapshot
test('HowToPlay render to match snapshot', () => {
  // render
  const howTo = renderer.create(<HowToPlay />).toJSON();

  // expected results
  expect(howTo).toMatchSnapshot();
});

// Home render to match snapshot
test('Home render to match snapshot', () => {
  const socket = new MockedSocket();
  const newGame = true;
  const first_player = 'A';
  const second_player = 'B';
  const gameFinished = false;
  // render
  const home = renderer
    .create(
      <Home
        socket={socket}
        first_player={first_player}
        second_player={second_player}
        newGame={newGame}
        gameFinished={gameFinished}
      />
    )
    .toJSON();

  // expected results
  expect(home).toMatchSnapshot();
});

test('Connect page renders correctly', () => {
  // render Form on virtual dom
  render(<Form />);

  // elements to test
  const name = screen.getByTestId('name');
  const name_label = screen.getByTestId('name-label');
  const email = screen.getByTestId('email');
  const email_label = screen.getByTestId('email-label');
  const comments = screen.getByTestId('comments');
  const comments_label = screen.getByTestId('comments-label');
  const submit = screen.getByTestId('submit');
  const reset = screen.getByTestId('reset');

  const handleSubmit = jest.fn();

  // interaction
  fireEvent.click(reset);
  userEvent.click(submit);

  // expected results
  expect(submit).toBeDefined();
  expect(reset).toBeDefined();

  expect(handleSubmit).toHaveBeenCalledTimes(1);

  expect(name_label).toHaveTextContent('Name');
  expect(email_label).toHaveTextContent('Email');
  expect(comments_label).toHaveTextContent('Comments');
});

// Connect render to match snapshot
test('Connect render to match snapshot', () => {
  // render
  const connect = renderer.create(<Form />).toJSON();

  // expected results
  expect(connect).toMatchSnapshot();
});

// Comments render to match snapshot
test('Comments render to match snapshot', () => {
  // render
  const comments = renderer.create(<Comments />).toJSON();

  // expected results
  expect(comments).toMatchSnapshot();
});

// Comments page renders correctly without any comments
test('Comments page renders corretly without any comments', () => {
  // render About on virtual dom
  render(<Comments />);

  // elements to test
  const header = screen.getByTestId('comments_header');
  const explain = screen.getByTestId('comments_explain');
  const display = screen.getByTestId('comments_display');

  // expected results
  expect(header).toHaveTextContent('Comments');
  expect(explain).toHaveTextContent(
    'You can leave us comments by filling out the form in the Connect page!'
  );
  expect(display).toHaveTextContent('No comments received');
});
