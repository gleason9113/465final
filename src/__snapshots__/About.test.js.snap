// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About render to match snapshot 1`] = `
<div
  className="container mx-auto text-center"
>
  <h1
    araia-label="about heading"
    className="about display-1"
    data-testid="about"
  >
    About
  </h1>
  <p
    data-testid="p1"
  >
    This is a CS465/565 Final Project by Ariel, Cera, and Alex
  </p>
  <p
    data-testid="p2"
  >
    We wanted to implement a game using Node.js, React.js, Express.js, and Socket.io
  </p>
  <p
    data-testid="p3"
  >
    We decided on a Tic-Tac-Toe game
  </p>
  <p
    data-testid="p4"
  >
    Our project code can be found in: 
  </p>
  <p>
    <a
      araia-label="github link code"
      data-testid="link-project"
      href="https://github.com/gleason9113/465final"
      rel="noreferrer"
      target="_blank"
    >
      Project
    </a>
  </p>
  <p>
    More about each of us can be found in:
  </p>
  <p>
    <a
      araia-label="github link cera"
      data-testid="cera-git"
      href="https://github.com/C3ra906"
      rel="noreferrer"
      target="_blank"
    >
      Cera's Github
    </a>
  </p>
  <p>
    <a
      araia-label="github link alex"
      data-testid="alex-git"
      href="https://github.com/sashetov"
      rel="noreferrer"
      target="_blank"
    >
      Alex's Github
    </a>
  </p>
  <p>
    <a
      araia-label="github link ariel"
      data-testid="ariel-git"
      href="https://github.com/gleason9113"
      rel="noreferrer"
      target="_blank"
    >
      Ariel's Github
    </a>
  </p>
</div>
`;

exports[`Board render to match snapshot when game is finished 1`] = `
<div
  className="container-sm w-50"
>
  <div
    aria-label="row-0"
    className="row"
  >
    <div
      aria-label="cell 0"
      className="col cell text-center"
      id="0"
      onClick={[Function]}
    />
    <div
      aria-label="cell 1"
      className="col cell text-center"
      id="1"
      onClick={[Function]}
    />
    <div
      aria-label="cell 2"
      className="col cell text-center"
      id="2"
      onClick={[Function]}
    />
  </div>
  <div
    aria-label="row-1"
    className="row text-center"
  >
    <div
      aria-label="cell 3"
      className="col cell text-center"
      id="3"
      onClick={[Function]}
    />
    <div
      aria-label="cell 4"
      className="col cell text-center"
      id="4"
      onClick={[Function]}
    />
    <div
      aria-label="cell 5"
      className="col cell text-center"
      id="5"
      onClick={[Function]}
    />
  </div>
  <div
    aria-label="row-2"
    className="row"
  >
    <div
      aria-label="cell 6"
      className="col cell text-center"
      id="6"
      onClick={[Function]}
    />
    <div
      aria-label="cell 7"
      className="col cell text-center"
      id="7"
      onClick={[Function]}
    />
    <div
      aria-label="cell 8"
      className="col cell text-center"
      id="8"
      onClick={[Function]}
    />
  </div>
  <div
    className="row"
    style={
      Object {
        "margin": "20px",
      }
    }
  />
  <div
    className="row"
  >
    <div
      aria-label="message board"
      className="col message text-center"
      id="user_message"
    />
  </div>
</div>
`;

exports[`Board render to match snapshot when game is not finished 1`] = `
<div
  className="container-sm w-50"
>
  <div
    aria-label="row-0"
    className="row"
  >
    <div
      aria-label="cell 0"
      className="col cell text-center"
      id="0"
      onClick={[Function]}
    />
    <div
      aria-label="cell 1"
      className="col cell text-center"
      id="1"
      onClick={[Function]}
    />
    <div
      aria-label="cell 2"
      className="col cell text-center"
      id="2"
      onClick={[Function]}
    />
  </div>
  <div
    aria-label="row-1"
    className="row text-center"
  >
    <div
      aria-label="cell 3"
      className="col cell text-center"
      id="3"
      onClick={[Function]}
    />
    <div
      aria-label="cell 4"
      className="col cell text-center"
      id="4"
      onClick={[Function]}
    />
    <div
      aria-label="cell 5"
      className="col cell text-center"
      id="5"
      onClick={[Function]}
    />
  </div>
  <div
    aria-label="row-2"
    className="row"
  >
    <div
      aria-label="cell 6"
      className="col cell text-center"
      id="6"
      onClick={[Function]}
    />
    <div
      aria-label="cell 7"
      className="col cell text-center"
      id="7"
      onClick={[Function]}
    />
    <div
      aria-label="cell 8"
      className="col cell text-center"
      id="8"
      onClick={[Function]}
    />
  </div>
  <div
    className="row"
    style={
      Object {
        "margin": "20px",
      }
    }
  />
  <div
    className="row"
  >
    <div
      aria-label="message board"
      className="col message text-center"
      id="user_message"
    />
  </div>
</div>
`;

exports[`ButtonHanddler render to match snapshot 1`] = `
<div>
  <div
    className="nav-container"
  >
    <button
      className="nav-btn"
      data-testid="home_button"
      id="home"
      onClick={[Function]}
      type="button"
    >
      Home
    </button>
    <button
      className="nav-btn"
      data-testid="about_button"
      id="about"
      onClick={[Function]}
      type="button"
    >
      About
    </button>
    <button
      className="nav-btn"
      data-testid="connect_button"
      id="connect"
      onClick={[Function]}
      type="button"
    >
      Connect
    </button>
    <button
      className="nav-btn"
      data-testid="howToPlay_button"
      id="how"
      onClick={[Function]}
      type="button"
    >
      How To Play
    </button>
    <button
      className="nav-btn"
      data-testid="comment_button"
      id="comments"
      onClick={[Function]}
      type="button"
    >
      Comments
    </button>
  </div>
  <div>
    <div
      className="container mx-auto text-center"
    >
      <h1
        className="display-1"
      >
        Tic-Tac-Toe
      </h1>
    </div>
    <form
      class="form w-50 mx-auto mt-5 p-3"
      id="player"
      method="get"
      onSubmit={[Function]}
    >
      <div
        class="form-group mx-auto my-2"
      >
        <label
          class="py-2"
          for="name"
        >
          Please enter player name:
        </label>
        <input
          class="form-control"
          id="name"
          name="name"
          type="text"
        />
      </div>
      <div
        class="row form-group mx-auto mb-2"
      >
        <input
          class="col btn btn-primary my-2 px-5"
          name="submit"
          type="submit"
          value="Play"
        />
        <input
          class="col btn btn-secondary mx-1 my-2 px-5"
          name="reset"
          type="reset"
          value="Reset"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`Buttons render to match snapshot 1`] = `
<div
  className="nav-container"
>
  <button
    className="nav-btn"
    data-testid="home_button"
    id="home"
    onClick={[MockFunction]}
    type="button"
  >
    Home
  </button>
  <button
    className="nav-btn"
    data-testid="about_button"
    id="about"
    onClick={[MockFunction]}
    type="button"
  >
    About
  </button>
  <button
    className="nav-btn"
    data-testid="connect_button"
    id="connect"
    onClick={[MockFunction]}
    type="button"
  >
    Connect
  </button>
  <button
    className="nav-btn"
    data-testid="howToPlay_button"
    id="how"
    onClick={[MockFunction]}
    type="button"
  >
    How To Play
  </button>
  <button
    className="nav-btn"
    data-testid="comment_button"
    id="comments"
    onClick={[MockFunction]}
    type="button"
  >
    Comments
  </button>
</div>
`;

exports[`Comments render to match snapshot 1`] = `
<div
  className="comments_container"
>
  <h1
    aria-label="comments header"
    className="comments_header"
    data-testid="comments_header"
  >
    Comments
  </h1>
  <p
    aria-label="how to post comments"
    className="comments_explain"
    data-testid="comments_explain"
  >
    You can leave us comments by filling out the form in the Connect page!
  </p>
  <pre>
    <div
      aria-label="comment board"
      className="comments_display"
      data-testid="comments_display"
    >
      No comments received
    </div>
  </pre>
</div>
`;

exports[`Connect render to match snapshot 1`] = `
<div
  className="container mx-auto my-auto text-center"
>
  <form
    aria-label="connect form"
    className="connect_form form w-50 mx-auto mt-5 p-3"
    data-testid="connect_form"
    onSubmit={[Function]}
  >
    <h1
      aria-label="connect header"
      className="mt-2 mb-4 connect_header"
      data-testid="connect_header"
    >
      Let's Connect!
    </h1>
    <div
      className="form-group mx-auto my-2"
    >
      <label
        aria-label="name label"
        className="py-2 name_label"
        data-testid="name-label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        aria-label="name field"
        className="form-control"
        data-testid="name"
        id="name"
        type="text"
      />
    </div>
    <div
      className="form-group mx-auto my-2"
    >
      <label
        aria-label="email label"
        className="py-2"
        data-testid="email-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        aria-label="email field"
        className="form-control"
        data-testid="email"
        id="email"
        type="email"
      />
    </div>
    <div
      className="form-group mx-auto my-2"
    >
      <label
        aria-label="comments label"
        className="py-2"
        data-testid="comments-label"
        htmlFor="comments"
      >
        Comments
      </label>
      <textarea
        aria-label="comments field"
        className="form-control"
        data-testid="comments"
        id="comments"
        rows="4"
      />
    </div>
    <div
      className="row form-group mx-auto mb-2"
    >
      <input
        aria-label="submit"
        className="col btn btn-primary my-2 px-5"
        data-testid="submit"
        name="submit"
        type="submit"
        value="Submit"
      />
      <input
        aria-label="reset"
        className="col btn btn-secondary mx-1 my-2 px-5"
        data-testid="reset"
        name="reset"
        type="reset"
        value="Reset"
      />
    </div>
  </form>
</div>
`;

exports[`Home render to match snapshot when newGame is false 1`] = `
<div>
  <div
    className="container mx-auto text-center"
  >
    <h1
      className="display-1"
    >
      Tic-Tac-Toe
    </h1>
  </div>
  <div
    className="container-sm w-50"
  >
    <div
      aria-label="row-0"
      className="row"
    >
      <div
        aria-label="cell 0"
        className="col cell text-center"
        id="0"
        onClick={[Function]}
      />
      <div
        aria-label="cell 1"
        className="col cell text-center"
        id="1"
        onClick={[Function]}
      />
      <div
        aria-label="cell 2"
        className="col cell text-center"
        id="2"
        onClick={[Function]}
      />
    </div>
    <div
      aria-label="row-1"
      className="row text-center"
    >
      <div
        aria-label="cell 3"
        className="col cell text-center"
        id="3"
        onClick={[Function]}
      />
      <div
        aria-label="cell 4"
        className="col cell text-center"
        id="4"
        onClick={[Function]}
      />
      <div
        aria-label="cell 5"
        className="col cell text-center"
        id="5"
        onClick={[Function]}
      />
    </div>
    <div
      aria-label="row-2"
      className="row"
    >
      <div
        aria-label="cell 6"
        className="col cell text-center"
        id="6"
        onClick={[Function]}
      />
      <div
        aria-label="cell 7"
        className="col cell text-center"
        id="7"
        onClick={[Function]}
      />
      <div
        aria-label="cell 8"
        className="col cell text-center"
        id="8"
        onClick={[Function]}
      />
    </div>
    <div
      className="row"
      style={
        Object {
          "margin": "20px",
        }
      }
    />
    <div
      className="row"
    >
      <div
        aria-label="message board"
        className="col message text-center"
        id="user_message"
      />
    </div>
  </div>
</div>
`;

exports[`Home render to match snapshot when newGame is true 1`] = `
<div>
  <div
    className="container mx-auto text-center"
  >
    <h1
      className="display-1"
    >
      Tic-Tac-Toe
    </h1>
  </div>
  <div
    className="container-sm w-50"
  >
    <div
      aria-label="row-0"
      className="row"
    >
      <div
        aria-label="cell 0"
        className="col cell text-center"
        id="0"
        onClick={[Function]}
      />
      <div
        aria-label="cell 1"
        className="col cell text-center"
        id="1"
        onClick={[Function]}
      />
      <div
        aria-label="cell 2"
        className="col cell text-center"
        id="2"
        onClick={[Function]}
      />
    </div>
    <div
      aria-label="row-1"
      className="row text-center"
    >
      <div
        aria-label="cell 3"
        className="col cell text-center"
        id="3"
        onClick={[Function]}
      />
      <div
        aria-label="cell 4"
        className="col cell text-center"
        id="4"
        onClick={[Function]}
      />
      <div
        aria-label="cell 5"
        className="col cell text-center"
        id="5"
        onClick={[Function]}
      />
    </div>
    <div
      aria-label="row-2"
      className="row"
    >
      <div
        aria-label="cell 6"
        className="col cell text-center"
        id="6"
        onClick={[Function]}
      />
      <div
        aria-label="cell 7"
        className="col cell text-center"
        id="7"
        onClick={[Function]}
      />
      <div
        aria-label="cell 8"
        className="col cell text-center"
        id="8"
        onClick={[Function]}
      />
    </div>
    <div
      className="row"
      style={
        Object {
          "margin": "20px",
        }
      }
    />
    <div
      className="row"
    >
      <div
        aria-label="message board"
        className="col message text-center"
        id="user_message"
      />
    </div>
  </div>
</div>
`;

exports[`HowToPlay render to match snapshot 1`] = `
<div
  className="howtoplay_container"
>
  <h1
    aria-label="How To Play Tic-Tac-Toe header"
    className="display-1 text-center howtoplay_header"
    data-testid="how_to_header"
  >
    How to Play Tic-Tac-Toe
  </h1>
  <p
    aria-label="require 2 players"
    data-testid="two_players"
  >
    This game is played with two players
  </p>
  <div>
    <p
      data-testid="rules"
    >
      Rules:
    </p>
    <ul
      aria-label="rules list"
      data-testid="list_of_rules"
    >
      <li>
        The first player enter their name to connect will be 'Player 1'
      </li>
      <li>
        The second player to enter their name to connect will be 'Player 2'
      </li>
      <li>
        Player 1 needs to wait until there is a Player 2
      </li>
      <li>
        Player 1 is assigned the symbol 'X' and Player 2 is assigned the symbol 'O'
      </li>
      <li>
        Once there are two players, the game screen will load the 3x3 board
      </li>
      <li>
        The game is played by clicking on a square of the board
      </li>
      <li>
        If the move is valid, the chosen square will be marked by the player's symbol
      </li>
      <li>
        Whoever is the first to connect 3 of their symbols horizontally, vertically, or diagonally wins the game
      </li>
      <li>
        The game will end in a tie if all squares are filled but no one has connected 3 of their symbols
      </li>
    </ul>
  </div>
  <div>
    <p
      aria-label="example wins"
      data-testid="wins"
    >
      Example Wins:
    </p>
    <pre
      aria-label="horizontal wins"
      data-testid="h_wins"
    >
      <p>
         X | X | X       |   |          |   |   
-----------   -----------    -----------
   |   |       X | X | X        |   |
-----------   -----------    -----------
   |   |         |   |        X | X | X 
      </p>
    </pre>
    <p />
    <pre
      aria-label="vertical wins"
      data-testid="v_wins"
    >
      <p>
         O |   |         | O |          |   | O 
-----------   -----------    -----------
 O |   |         | O |          |   | O
-----------   -----------    -----------
 O |   |         | O |          |   | O
      </p>
    </pre>
    <p />
    <pre
      aria-label="diagonal wins"
      data-testid="d_wins"
    >
      <p>
         X |   |         |   | O
-----------   -----------
   | X |         | O |   
-----------   -----------
   |   | X     O |   |    
      </p>
    </pre>
  </div>
  <div>
    <p
      aria-label="example tied"
      data-testid="ties"
    >
      Example Tied Conditions:
    </p>
    <pre
      aria-label="tied game"
      data-testid="tied_ex"
    >
      <p>
         X | O | X     X | O | O      X | X | O
-----------   -----------    -----------
 X | X | O     X | O | X      O | O | X
-----------   -----------    -----------
 O | O | X     O | X | X      X | X | O 
      </p>
    </pre>
  </div>
  <div>
    <p
      aria-label="enjoy"
      data-testid="enjoy"
    >
      Hope you enjoy the game!
    </p>
  </div>
</div>
`;
